contrib/fixclock-hook

```diff
@@ -26,27 +28,96 @@ You can change the NTP server like this:
 Unfortunately, something went wrong...
 
 $ERROR
+"""
+
+LXC_TPL = f"""\
+##########################
+## FIXCLOCK HOOK FAILED ##
+##########################
+
+Amazon S3 and Duplicity need a UTC synchronized clock, but TKLBAM detected
+that we're running in a container ($CONTAINER).
 
+As the container host manages the local time (and an unprivileged container
+can't update it) the automated time update performed by TKLBAM will be
+skipped.
+
+To ensure that TKLBAM functions properly, please ensure that your host OS
+is regularly updating it's clock. If you're using Proxmox, then this
+should already be happening automatically.
+
+If your host is regularly updating it's clock, then you can safely ignore
+this warning. To stop this message from displaying, please run:
+
+    chmod -x {realpath(sys.argv[0])}
 """
 
-def fixclock():
-    os.environ['PATH'] += ':/usr/sbin'
 
-    command = ["ntpdate", "-u", NTPSERVER]
+def echo(msg: str, stderr: bool = False, fatal: bool = False) -> None:
+    """ Echo messge to terminal (exit 1 if fatal)"""
+    output = sys.stdout
+    if stderr:
+        output = sys.stderr
+    print(msg, end=' ', file=output)
+    print(file=output)
+    if fatal:
+        sys.exit(1)
+
 
+def runit(command: list[str]) -> tuple[int, str, str]:
+    """Run 'command' via subprocess; returns:
+
+        tuple(returncode, stdout, stderr)
+    """
+    if '/usr/sbin' not in os.environ['PATH']:
+        os.environ['PATH'] += ':/usr/sbin'
+    full_path = shutil.which(command[0])
+    if full_path is None:
+        echo(f"FATAL[{__file__}]: Couldn't find path to {command[0]}",
+             stderr=True, fatal=True)
     p = subprocess.run(command, capture_output=True, text=True)
-    if p.returncode != 0:
+    return p.returncode, p.stdout, p.stderr
+
+
+def is_container() -> str:
+    """Check if running in container, if yes, return container type,
+    otherwise 'no'"""
+
+    returncode, stdout, stderr = runit(['systemd-detect-virt', '-c'])
+
+    if stdout.strip() == 'none':
```

Comment:
exit code should be checked here instead, and then if it's 0, the container type can be read

> If a virtualization technology is detected, 0 is returned, a non-zero code otherwise.

https://www.freedesktop.org/software/systemd/man/latest/systemd-detect-virt.html